test/blackbox-tests/test-cases/dir-target-dep.t/run.t

```diff
@@ -36,8 +36,8 @@ Dune crashes if there's a file named after the directory target
   1 | (rule
   2 |  (targets dir)
   3 |  (action (bash "mkdir %{targets}")))
```

Comment:
I noticed that Dune allows a rule to produce a directory for `(targets dir)`, which is an overlap in functionality with new directory targets. Do we want to do anything about this? We could require that all non-`dir` targets must be files which seems more appropriate, but perhaps somebody depends on the current behaviour...

Reply:
This comes from #1574, which was a minimal change to kind of support directory targets. This feature was never documented as it wasn't finished. Now that we properly support directory targets, we can remove the old undocumented and unfinished support. Though it would be nice if we could still detect it and suggest to the user to use a directory target, so that existing users of this feature can move forward more easily.

Reply:
Are you sure that's the right ticket? The diff doesn't seem to include anything relevant. 

Good to know that this is undocumented and we don't guarantee stability. I'll try to turn all such cases into errors and suggest using proper directory targets instead.

Reply:
Indeed, it was meant to be #1547.

Reply:
Aha, thanks, that looks right!